app/validators/embargo_date_parts.rb

```diff
@@ -0,0 +1,14 @@
+# typed: true
+# frozen_string_literal: true
+
+# Validates embargo month and day
+class EmbargoDateParts < ActiveModel::Validator
+  def validate(record)
+    # If we've already successfully deserialized the date, there's no point to continue with this
+    return if record.embargo_date.instance_of?(Date)
+
+    month = record.send('embargo_date(2i)')
```

Comment:
Are these ever not nil?  Haven't they been deleted in EmbargoDateParams filter?

Reply:
Changed to just check if `embargo_date` is nil